larq/quantizers.py

```diff
@@ -194,30 +194,37 @@ def __init__(self, clip_value=1.0):
         super().__init__(ste_sign, clip_value=clip_value)
 
 
+def _scaled_sign(x):
+    return 1.3 * ste_sign(x)
+
+
 @utils.register_keras_custom_object
 @utils.set_precision(1)
-def magnitude_aware_sign(x, clip_value=1.0):
+def magnitude_aware_sign(x, clip_value=1.0, channel_wise_scale_factor=True):
     r"""
     Magnitude-aware sign for Bi-Real Net.
 
     ```plot-activation
-    quantizers.magnitude_aware_sign
+    quantizers._scaled_sign
     ```
 
     # Arguments
     x: Input tensor
     clip_value: Threshold for clipping gradients.
+    channel_wise_scale_factor: True if the scale factor should be calculated channel wise,
+        False for a layer wise scale factor.
 
     # Returns
-    Scaled binarized tensor (with values in $\{-a, a\}$, where $a$ is a float).
+    Scaled binarized tensor (with values in \\(\\{-a, a\\}\\), where $a$ is a float).
 
     # References
     - [Bi-Real Net: Enhancing the Performance of 1-bit CNNs With Improved
       Representational Capability and Advanced Training
       Algorithm](https://arxiv.org/abs/1808.00278)
 
     """
-    scale_factor = tf.reduce_mean(tf.abs(x), axis=list(range(len(x.shape) - 1)))
+    axis = list(range(len(x.shape) - 1)) if channel_wise_scale_factor else None
+    scale_factor = tf.reduce_mean(tf.abs(x), axis=axis)
```

Comment:
True, this seems indeed to be the same. I think I changed my mind now that I have seen how it is used in the xnornet model.  It might be better to just add it in a similar way to the model file. I will remove this commit again.